### PR TITLE
perf(replays): instrument replay segment downloads

### PR DESCRIPTION
**Pull request by JoshFerge**
- adds instrumentation to our download segments function. is currently not instrumented as we stream the HTTP response, and our transaction for that endpoint ends before this function runs. 

example transaction that this generates: https://sentry-emerging-tech.sentry.io/performance/josh-testing-ee:deb5f42458314d0e9e41356769158914/?project=4505015879598080&query=http.method%3AGET&referrer=performance-transaction-summary&statsPeriod=14d&transaction=ProjectReplayRecordingSegmentIndexEndpoint.download_segments&unselectedSeries=p100%28%29

### CODE REVIEW

**Comment by JoshFerge** [comment]
if python sdk folks have an idea how to have this instead be a span under the transaction of this endpoint https://github.com/getsentry/sentry/blob/6c39a0ffacb84c2f1e3062342377933b6dfe92fc/src/sentry/replays/endpoints/project_replay_recording_segment_index.py#L28 , or link them via trace_id, would be interested!

**Comment by JoshFerge** [comment]
typing issue: 
```
src/sentry/replays/usecases/reader.py:249: error: Item "NoOpSpan" of "Union[Transaction, NoOpSpan]" has no attribute "set_measurement"
```

going to ignore for now.

File: src/sentry/replays/usecases/reader.py [hunk 1]
```diff
@@ -238,9 +241,22 @@ def _fetch_segments_from_snuba(
...
     """Download segment data from remote storage."""
+
+    # start a sentry transaction to pass to the thread pool workers
+    transaction = sentry_sdk.start_transaction(
+        op="http.server",
+        name="ProjectReplayRecordingSegmentIndexEndpoint.download_segments",
+    )
+    transaction.set_measurement("number_of_segments", len(segments))  # type: ignore
+
+    current_hub = sentry_sdk.Hub(sentry_sdk.Hub.current)
```
Comment (sl0thentr0py) [inline src/sentry/replays/usecases/reader.py:252]: you don't need this copy I guess

**Review by sl0thentr0py — APPROVED** [review]

File: src/sentry/replays/usecases/reader.py [hunk 2]
```diff
@@ -238,9 +241,22 @@ def _fetch_segments_from_snuba(
 
 def download_segments(segments: List[RecordingSegmentStorageMeta]) -> Iterator[bytes]:
     """Download segment data from remote storage."""
+
+    # start a sentry transaction to pass to the thread pool workers
+    transaction = sentry_sdk.start_transaction(
```
Comment (sl0thentr0py) [inline src/sentry/replays/usecases/reader.py:246]: if you want these to be spans under the request transaction, just don't start this transaction here and then just do `with sentry_sdk.start_span(..)` below

Comment (JoshFerge) [reply src/sentry/replays/usecases/reader.py:246]: i assume the sentry span gets finished when the django endpoint returns (we start streaming the response) -- not sure what would happen if i did that, none of the current child spans end up in the endpoint transaction now e.g. https://sentry.sentry.io/performance/sentry:3a5b5275f5fb4206924a9f14b875c965/?project=1&query=http.method%3AGET&statsPeriod=14d&transaction=%2Fapi%2F0%2Fprojects%2F%7Borganization_slug%7D%2F%7Bproject_slug%7D%2Freplays%2F%7Breplay_id%7D%2Frecording-segments%2F

Comment (sl0thentr0py) [reply src/sentry/replays/usecases/reader.py:246]: hmm ok I've never tried streaming stuff while testing the SDK, but yea if nothing's waiting on the thread pool stuff to finish, the parent transaction has no way of ensuring timings, so then technically, a new transaction makes sense.

However, dynamic sampling, tracing etc is broken if you just call `start_transaction` right now, you'd need to do the whole [`continue_from_headers`](https://github.com/getsentry/sentry-python/blob/c147d7129f82845a69a2dba68b0e62598dbfe295/sentry_sdk/tracing.py#L271-L275) dance. This is something we're aware is broken and not user friendly (transaction within transactions within a single SDK).

**Pull request merged**